modules/management-console-agent/src/test/java/org/apache/ignite/agent/processor/action/DistributedActionProcessorTest.java

```diff
@@ -0,0 +1,246 @@
+        boolean responsesHasCorrectConsistentIds = allNodeConsistentIds.containsAll(
+            responses
+                .stream()
+                .map(JobResponse::getNodeConsistentId)
+                .collect(toSet())
+        );
+
+        assertTrue(responsesHasCorrectConsistentIds);
+    }
+
+    /**
+     * Should execute action on all nodes with one node stop after 1 second.
+     */
+    @Test
+    public void shouldExecuteActionOnAllNodesWithNodeStop() throws Exception {
+        Request req = new Request()
+            .setId(UUID.randomUUID())
+            .setAction("IgniteTestActionController.nodeIdActionWithSleep")
+            .setArgument(5000);
+
+        executeActionAndStopNode(req, 1000, 1, res -> {
+            List<TaskResponse> taskResults =
+                interceptor.getAllPayloads(buildActionTaskResponseDest(cluster.id(), req.getId()), TaskResponse.class);
+
+            Optional<TaskResponse> runningTask = taskResults.stream().filter(r -> r.getStatus() == RUNNING).findFirst();
+            Optional<TaskResponse> failedTask = taskResults.stream().filter(r -> r.getStatus() == FAILED).findFirst();
+
+            if (runningTask.isPresent() && failedTask.isPresent()) {
+                long failedJobCnt = res.stream()
+                    .filter(r -> r.getStatus() == FAILED)
+                    .count();
+
+                return res.size() == failedTask.get().getJobCount() && failedJobCnt == 1;
+            }
+
+            return false;
+        });
+    }
+
+    /**
+     * Should send error response on response with invalid node id.
+     */
+    @Test
+    public void shouldSendErrorResponseWithInvalidNodeId() {
+        Request req = new Request()
+            .setId(UUID.randomUUID())
+            .setAction("IgniteTestActionController.nodeIdAction")
+            .setNodeIds(singleton(UUID.randomUUID()));
+
+        executeAction(req, (res) -> {
+            JobResponse r = F.first(res);
+            TaskResponse taskRes =
+                interceptor.getPayload(buildActionTaskResponseDest(cluster.id(), req.getId()), TaskResponse.class);
+
+            return taskRes.getStatus() == FAILED && r.getStatus() == FAILED && r.getError().getCode() == INTERNAL_ERROR_CODE;
+        });
+    }
+
+    /**
+     * Should send error response on response with invalid argument.
+     */
+    @Test
+    public void shouldSendErrorResponseWithInvalidArgument() {
+        Request req = new Request()
+            .setId(UUID.randomUUID())
+            .setAction("BaselineActions.updateAutoAdjustAwaitingTime")
+            .setArgument("value");
+
+        executeAction(req, (res) -> {
+            JobResponse r = F.first(res);
+            TaskResponse taskRes =
+                interceptor.getPayload(buildActionTaskResponseDest(cluster.id(), req.getId()), TaskResponse.class);
+
+            return taskRes.getStatus() == FAILED && r.getError().getCode() == PARSE_ERROR_CODE;
```

Comment:
Don't we need ```&& r.getStatus() == FAILED ``` here as well?

Reply:
If we have error code then job result has the FAILED status :)

Reply:
@smoldachev despite that's an offtop, is that covered in some other test?

Reply:
You can look at org.apache.ignite.agent.processor.action.DistributedActionProcessorTest#shouldSendErrorResponseWithInvalidNodeId

There I check such case :)